src/concrete/ml/quantization/post_training.py

```diff
@@ -23,6 +23,55 @@
 from .quantized_ops import QuantizedBrevitasQuant
 from .quantizers import QuantizationOptions, QuantizedArray, UniformQuantizer
 
+# pylint: disable=too-many-lines
+
+
+def get_n_bits_dict_trees(n_bits: Union[int, Dict[str, int]]) -> Dict[str, int]:
+    """Convert the n_bits parameter into a proper dictionary for tree based-models.
+
+    Args:
+        n_bits (int, Dict[str, int]): number of bits for quantization, can be a single value or
+            a dictionary with the following keys :
+            - "op_inputs" (mandatory)
+            - "op_leaves" (optional)
+            When using a single integer for n_bits, its value is assigned to "op_inputs" and
+            "op_leaves" bits.
+
+    Returns:
+        n_bits_dict (Dict[str, int]): A dictionary properly representing the number of bits to use
+            for quantization.
+    """
+
+    assert_true(
+        isinstance(n_bits, int)
+        or (isinstance(n_bits, Dict) and set(n_bits.keys()).issubset({"op_inputs", "op_leaves"})),
+        "Invalid n_bits, either pass an integer or a dictionary containing integer values for "
+        "the following keys:\n"
+        "- `op_inputs` and `op_leaves` (mandatory)",
+    )
+
+    n_bits_dict: Dict = {}
+
+    # If a single integer is passed, we use a default value for the model's input and
+    # output bits
+    if isinstance(n_bits, int):
+        n_bits_dict = {
+            "op_inputs": n_bits,
+            "op_leaves": n_bits,
+        }
+
+    elif isinstance(n_bits, Dict):
+        n_bits_dict.update(n_bits)
+
+    assert_true(
```

Comment:
same here maybe this should be an exception raise instead of an assert

Reply:
you mean you'll do it in a following PR like you said above right ?

Reply:
yes